Yoast/wordpress-seo
# Optimize breadcrumbs with indexables

## Summary

This is the first step in rewriting breadcrumbs to use indexables. This adds a new table that holds the entire hierarchy of an indexable, excluding crumbs that are set through global options.

It currently doesn't do anything with this table yet, it merely fills it.

This PR can be summarized in the following changelog entry:

* N/A

## Relevant technical choices:

* To avoid further increasing the size of the indexable_repository a new indexable_builder class has been introduced that wraps all the specific builders.

## Test instructions

This PR can be tested by following these steps:

* Create a page with a parent. You should see entries in the `wp_yoast_indexable_hierarchy` table that matches the indexables for that page and it's parent.
* Enable breadcrumbs and define a main taxonomy for posts as category.
* Create two categories a parent and a child. Set the parent to be the parent of the child.
* Create a post which has the child as it's category.
* There should be two entries in the hierarchy table matching both the child ( depth = 1 ) and the parent ( depth = 2 ).
* Create a post with multiple categories and mark one as the primary category.
* That category should be reflected in the hierarchy table.
* Create a post with multiple categories, one with a parent and one without. Do not mark any as the primary category.
* The category with the parent should be reflected in the hierarchy table.

## UI changes

## Documentation

## Quality assurance
